spring-xd-rest-client/src/main/java/org/springframework/xd/rest/client/impl/StreamTemplate.java

```diff
@@ -95,4 +97,17 @@ public void destroyAll() {
 		restTemplate.delete(resources.get("streams/definitions").expand());
 	}
 
+	@Override
+	public void deleteQueue(String name, Map<String, String> properties) {
+		String rootUri = resources.get("streams/clean/rabbit").toString() + "/{name}";
+		UriComponentsBuilder builder = UriComponentsBuilder.fromUriString(rootUri);
+
+		for(Map.Entry<String, String> entry: properties.entrySet()) {
+			builder.queryParam(entry.getKey(), entry.getValue());
+		}
```

Comment:
1. Happy to do this, but it seems odd to require all users to provide their RabbitMQ details given that they may not want this feature (and presumably so far few or none have used it). Let me know if you want me to go ahead with the change anyway (for stream and job).

2. Similarly happy to do this - I was trying to stick with the pattern set by the two classes, which keep their methods separate despite such commonality (e.g. deploy/undeploy/destroy etc.) Can you confirm that you want me to break that pattern?